kratos/geometries/nurbs_curve_on_surface_geometry.h

```diff
@@ -0,0 +1,294 @@
+        geometry's points.*/
+    typedef  typename BaseType::PointsArrayType PointsArrayType;
+
+    typedef  typename BaseType::CoordinatesArrayType CoordinatesArrayType;
+
+    /// Counted pointer of NurbsCurveOnSurfaceGeometry
+    KRATOS_CLASS_POINTER_DEFINITION(NurbsCurveOnSurfaceGeometry);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    NurbsCurveOnSurfaceGeometry(
+        typename NurbsSurfaceType::Pointer pSurface,
+        typename NurbsCurveType::Pointer pCurve)
+        : BaseType(pCurve->Points(), &msGeometryData), mpNurbsSurface(pSurface), mpNurbsCurve(pCurve)
+    {
+    }
+
+    /// Copy constructor.
+    NurbsCurveOnSurfaceGeometry(NurbsCurveOnSurfaceGeometry const& rOther)
+        : BaseType(rOther)
+        , mpNurbsSurface(rOther.mpNurbsSurface)
+        , mpNurbsCurve(rOther.mpNurbsCurve)
+    {
+    }
+
+    /// Copy constructor from a geometry with different point type.
+    template<class TOtherCurveContainerPointType, class TOtherSurfaceContainerPointType> NurbsCurveOnSurfaceGeometry(
+        NurbsCurveOnSurfaceGeometry<TWorkingSpaceDimension, TCurveContainerPointType, TOtherSurfaceContainerPointType> const& rOther)
+        : BaseType(rOther, &msGeometryData)
+        , mpNurbsSurface(rOther.mpNurbsSurface)
+        , mpNurbsCurve(rOther.mpNurbsCurve)
+    {
+    }
+
+    /// Destructor
+    ~NurbsCurveOnSurfaceGeometry() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
+     * Assignment operator.
+     *
+     * @note This operator don't copy the points and this
+     * geometry shares points with given source geometry. It's
+     * obvious that any change to this geometry's point affect
+     * source geometry's points too.
+     *
+     * @see Clone
+     * @see ClonePoints
+     */
+    NurbsCurveOnSurfaceGeometry& operator=(const NurbsCurveOnSurfaceGeometry& rOther)
+    {
+        BaseType::operator=(rOther);
+        mpNurbsSurface = rOther.mpNurbsSurface;
+        mpNurbsCurve = rOther.mpNurbsCurve;
+        return *this;
+    }
+
+    /**
+     * @brief Assignment operator for geometries with different point type.
+     *
+     * @note This operator don't copy the points and this
+     * geometry shares points with given source geometry. It's
+     * obvious that any change to this geometry's point affect
+     * source geometry's points too.
+     *
+     * @see Clone
+     * @see ClonePoints
+     */
+    template<class TOtherCurveContainerPointType, class TOtherSurfaceContainerPointType>
+    NurbsCurveOnSurfaceGeometry& operator=(
+        NurbsCurveOnSurfaceGeometry<TWorkingSpaceDimension, TOtherCurveContainerPointType, TOtherSurfaceContainerPointType> const & rOther)
+    {
+        BaseType::operator=(rOther);
+        mpNurbsSurface = rOther.mpNurbsSurface;
+        mpNurbsCurve = rOther.mpNurbsCurve;
+        return *this;
+    }
+
+    ///@}
+    ///@name OperationsZeroVector(3);
+    ///@{
+
+    /*typename BaseType::Pointer Create(
```

Comment:
This create function is required.

Reply:
This function expects as arguments PointsArrayType const& ThisPoints to create the geometry but for this type of geometry I don't know which points to pass, the ones of the surface or the ones of the curve? And if it is those of the curve, the ones in the parametric or the ones the physical space? Need opinions!

Reply:
The PointsArrayType is defined by the BaseType. This is essentially the surface points. For you, here it does not make a difference.